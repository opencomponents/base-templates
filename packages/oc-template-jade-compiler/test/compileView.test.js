jasmine.DEFAULT_TIMEOUT_INTERVAL = 50000;

const path = require('path');
const fs = require('fs-extra');
const compileView = require('../lib/compileView.js');

const componentPath = path.join(__dirname, '../../../mocks/jade-component');
const publishPath = path.join(componentPath, '_package');
const publishFileName = 'template.js';

const options = {
  componentPackage: {
    oc: {
      files: {
        template: {
          src: 'template.jade',
          type: 'jade'
        }
      }
    }
  },
  componentPath,
  publishPath,
  publishFileName
};

test('Should correctly compile the view', done => {
  compileView(options, (err, compiledViewInfo) => {
    const viewHashKey = compiledViewInfo.template.hashKey;
    compiledViewInfo.template.hashKey = 'dummyData';
    expect(compiledViewInfo).toMatchSnapshot();
    expect(
      fs
        .readFileSync(path.join(publishPath, publishFileName), 'UTF8')
        .replace(viewHashKey, 'dummyData')
    ).toMatchSnapshot();
    fs.removeSync(path.join(publishPath, publishFileName));
    done();
  });
});
