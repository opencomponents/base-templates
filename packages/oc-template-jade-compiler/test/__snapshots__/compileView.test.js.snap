// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should correctly compile the view 1`] = `
Object {
  "template": Object {
    "hashKey": "dummyData",
    "src": "template.js",
    "type": "jade",
  },
}
`;

exports[`Should correctly compile the view 2`] = `"var oc=oc||{};oc.components=oc.components||{},oc.components[\\"dummyData\\"]=function(e){var o=[];return o.push(\\"<h1>\\"+jade.escape(null==\\"Hello jade!\\"?\\"\\":\\"Hello jade!\\")+\\"</h1><style>h1{color:red;font-size:100px}</style>\\"),o.join(\\"\\")};"`;
